src/backend/distributed/planner/merge_planner.c

```diff
@@ -243,14 +243,19 @@ CreateNonPushableMergePlan(Oid targetRelationId, uint64 planId, Query *originalQ
 
 	CitusTableCacheEntry *targetRelation = GetCitusTableCacheEntry(targetRelationId);
 
-	/*
-	 * Get the index of the column in the source query that will be utilized
-	 * to repartition the source rows, ensuring colocation with the target
-	 */
-	distributedPlan->sourceResultRepartitionColumnIndex =
-		SourceResultPartitionColumnIndex(mergeQuery,
```

Comment:
@paragikjain  Remind me, we were failing in `SourceResultPartitionColumnIndex`because the target is not distributed? I thought we were failing in run time?

Reply:
Yes, we were failing in planning phase only as target is not distributed.